test_markdown_writer.py

```diff
@@ -324,3 +333,106 @@ def test_writes_markdown_file_with_non_hidden_columns_only(self):
         )
         self.assertEqual(content, expected_content)
         os.remove("issue_metrics.md")
+
+
+class TestWriteToMarkdownWithOutputCustomization(unittest.TestCase):
+    """Test the write_to_markdown function with the REPORT_TITLE environment variables set."""
+
+    def setUp(self):
+        os.environ["REPORT_TITLE"] = "My Custom Report Title"
+
+    def tearDown(self):
+        os.environ.pop("REPORT_TITLE")
+
+    def test_writes_markdown_file_with_output_customization(self):
+        """
+        Test that write_to_markdown writes the correct
+        markdown file.
+        """
+
+        # Create mock data
+        issues_with_metrics = [
+            IssueWithMetrics(
+                "Issue 1",
+                "https://github.com/user/repo/issues/1",
+                "alice",
+                timedelta(days=1),
+                timedelta(days=2),
+                timedelta(days=3),
+                {"bug": timedelta(days=1)},
+            ),
+            IssueWithMetrics(
+                "feat| Issue 2",  # title contains a vertical bar
+                "https://github.com/user/repo/issues/2",
+                "bob",
+                timedelta(days=3),
+                timedelta(days=4),
+                timedelta(days=5),
+                {"bug": timedelta(days=2)},
+            ),
+        ]
+        average_time_to_first_response = {
+            "avg": timedelta(days=2),
+            "med": timedelta(days=2),
+            "90p": timedelta(days=2),
+        }
+        average_time_to_close = {
+            "avg": timedelta(days=3),
+            "med": timedelta(days=3),
+            "90p": timedelta(days=3),
+        }
+        average_time_to_answer = {
+            "avg": timedelta(days=4),
+            "med": timedelta(days=4),
+            "90p": timedelta(days=4),
+        }
+        average_time_in_labels = {
+            "avg": {"bug": "1 day, 12:00:00"},
+            "med": {"bug": "1 day, 12:00:00"},
+            "90p": {"bug": "1 day, 12:00:00"},
+        }
+
+        num_issues_opened = 2
+        num_issues_closed = 1
+
+        # Call the function
+        write_to_markdown(
+            issues_with_metrics=issues_with_metrics,
+            average_time_to_first_response=average_time_to_first_response,
+            average_time_to_close=average_time_to_close,
+            average_time_to_answer=average_time_to_answer,
+            average_time_in_labels=average_time_in_labels,
+            num_issues_opened=num_issues_opened,
+            num_issues_closed=num_issues_closed,
+            labels=["bug"],
+            report_title="My Custom Report Title",
```

Comment:
@zkoppert I applied the suggested changes. That means moving the default value for the report title into the write_to_markdown() signature. It does lead to a better code structure overall, I think.

However one issue remains in this area:
We feed in the custom title as a parameter here.
Therefore we are not actually testing if it is passed through from the `REPORT_TITLE` env variable correctly.

Btw the same issue exists in the testing logic for the `hide_label_metrics` in [row 305 above](https://github.com/github/issue-metrics/pull/160/files#diff-48df48b606f1dcedafbeb2eea92427976f1c679d8b5d97de0d8211eb7610126dL305).
There I am not sure either if this test really covers what it should.

Reply:
I think overall I am a bit unclear about the intended logic of using ENV vars in the code.

Some ENV vars are read in various functions in `issue_metrics.py` - via `main()`, `get_env_vars()`, and others.
Yet again other ENV vars  are read directly in `markdown_writer.py`.

Reply:
Ideally we would consolidate all env variables being read from the get_env_vars() function.

Reply:
For this PR it would be good to follow that pattern. We can open a separate PR to get everything using the proper `get_env_vars()` function.